### PR TITLE
Fix catalogs not loading

### LINKED ISSUE #103: Catalog not loading for iGame
There are two related issues preventing iGame from running localized:

1. The file name of the catalog descriptor is used when generating the catalog source files to determine the file name of the catalog file that is actually opened. Since the catalog descriptor is `iGameStrings.cd` the generated `iGameStrings_cat.c` code for opening the catalog looks for `iGameStrings.catalog` rather than `iGame.catalog` as expected.
2. With VBCC and the currently generated code, the catalog file is never actually opened. With GCC (and DICE C) the functions for opening and closing the catalog are set up to be automatically called before and after `main()` using those compilers' features, but while VBCC supports similar functionality the functions aren't set up for that.

### PR DESCRIPTION
**Pull request by coldacid**
Builds upon PR #93 to properly support catalogs for localization in iGame. This renames the catalog descriptor file so that the generated code will look for the proper catalog file at runtime, and updates the code template to ensure that the open and close catalog functions are called before `main()` is called and after it returns (or `exit()` is called).

Fixes #103

### CODE REVIEW

**Comment by walkero-gr** [comment]
@coldacid Today I merged the code that has to do with drone CI/CD. Now we have a full automated way to compile all the versions of iGame on every PR and whenever we merge code on master or we create a release tag.
Unfortunately this made this PR to have a conflict on Makefile.docker. Sorry for that.

**Comment by coldacid** [comment]
@walkero-gr Not a problem, nothing a merge or rebase won't fix.

File: src/iGame.cd [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#language english
```
Comment (walkero-gr) [inline src/iGame.cd:1]: @coldacid should this CD file be generated automatically by the code, having all the latest updates of the strings?

Comment (coldacid) [reply src/iGame.cd:1]: It's the other way around, the code is generated from the CD file.

Comment (walkero-gr) [reply src/iGame.cd:1]: You are right... Thanks for that.

**Comment by walkero-gr** [comment]
@coldacid Is there more work you need to do on this PR or can we merge it?

**Comment by coldacid** [comment]
@walkero-gr I'm pretty sure this can be merged.

**Comment by coldacid** [comment]
@walkero-gr I'm going to rebase this and the `flexcat` branches on top of the v2.0 commit to make my life a little less terrible for integrating the Italian catalogue into the build and package process (and so that the history view in gitk is a little more sane). Please hold off on merging this or #93 until I'm done.

**Comment by coldacid** [comment]
@walkero-gr I'm done rebasing and validating.

**Comment by walkero-gr** [comment]
@coldacid Please, have a look on these conflicts so I can merge it. Thank you.

**Comment by coldacid** [comment]
@walkero-gr sorry for the delay, should be good now.

**Comment by coldacid** [comment]
@walkero-gr until this is merged I can't continue on the work for integrating catalogs into the build and package process.

**Pull request merged**